huaweicloud/services/cts/resource_huaweicloud_cts_tracker.go

```diff
@@ -176,8 +194,16 @@ func resourceCTSTrackerUpdate(ctx context.Context, d *schema.ResourceData, meta
 	// update other configurations
 	if d.IsNewResource() || d.HasChangeExcept("enabled") {
 		obsInfo := cts.TrackerObsInfo{
-			BucketName:     utils.String(d.Get("bucket_name").(string)),
-			FilePrefixName: utils.String(d.Get("file_prefix").(string)),
+			BucketName:      utils.String(d.Get("bucket_name").(string)),
+			FilePrefixName:  utils.String(d.Get("file_prefix").(string)),
+			IsSortByService: utils.Bool(d.Get("is_sort_by_service").(bool)),
+		}
+		if v, ok := d.GetOk("compress_type"); ok {
+			compressType := cts.GetTrackerObsInfoCompressTypeEnum().GZIP
+			if v.(string) != "gzip" {
+				compressType = cts.GetTrackerObsInfoCompressTypeEnum().JSON
+			}
```

Comment:
how about add a validation to the param of compress_type, if the value is not gzip, then the value will be json, no matter whether the value is json